pandas/tseries/tdi.py

```diff
@@ -438,23 +439,20 @@ def to_pytimedelta(self):
     def astype(self, dtype):
         dtype = np.dtype(dtype)
 
-        if dtype == np.object_:
+        if is_object_dtype(dtype):
             return self.asobject
-        elif dtype == _INT64_DTYPE:
-            return self.asi8.copy()
-        elif dtype == _TD_DTYPE:
+        elif is_timedelta64_ns_dtype(dtype):
             return self
```

Comment:
Just to confirm:  `.astype('m8[ns]')` should return a copy? 
It originally returned `self`. 
(At first, I thought that I messed it up.)